onnxruntime/core/platform/posix/env.cc

```diff
@@ -194,10 +194,16 @@ class PosixEnv : public Env {
     return std::thread::hardware_concurrency();
   }
 
-  std::vector<size_t> GetThreadAffinityMasks() const override {
-    std::vector<size_t> ret(std::thread::hardware_concurrency() / 2);
-    std::iota(ret.begin(), ret.end(), 0);
-    return ret;
+  size_t GetDefaultThreadpoolSetting(std::vector<size_t>& affinities) const override {
+    affinities.clear();
+    affinities.resize(std::thread::hardware_concurrency() >> 1);
```

Comment:
>std::thread::hardware_concurrency() >> 1 [](http://example.com/codeflow?start=22&length=40)

What happens if the return value is 1?

Reply:
The vector affinities will be of size 0, this will cause  CreateThreadPoolHelper(...) in thread_utils.cc report an error and return with a null unique_ptr.